File: app/src/main/java/com/hjc/base/ui/home/adapter/ArticleAdapter.java
package com.hjc.base.ui.home.adapter;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;
import com.hjc.base.R;
import com.hjc.base.databinding.ItemArticleBinding;
import com.hjc.base.bean.ArticleBean;

import org.jetbrains.annotations.NotNull;

public class ArticleAdapter extends BaseQuickAdapter<ArticleBean.DataBean.DatasBean, BaseViewHolder> {

    public ArticleAdapter() {
        super(R.layout.item_article);
    }

    @Override
    protected void onItemViewHolderCreated(@NotNull BaseViewHolder viewHolder, int viewType) {
        DataBindingUtil.bind(viewHolder.itemView);
    }

    @Override
    protected void convert(@NonNull BaseViewHolder helper, ArticleBean.DataBean.DatasBean item) {
        if (item == null) {
            return;
        }

        ItemArticleBinding binding = helper.getBinding();
        if (binding != null) {
            binding.setArticleBean(item);
        }
    }
}

File: app/src/main/java/com/hjc/base/adapter/ArticleAdapter.kt
package com.hjc.base.adapter

import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.viewholder.BaseViewHolder
import com.hjc.base.R
import com.hjc.base.databinding.ItemArticleBinding
import com.hjc.base.model.ArticleBean

/**
 * @Author: HJC
 * @Date: 2021/1/8 16:22
 * @Description: 文章Adapter
 */
class ArticleAdapter(data: MutableList<ArticleBean>?) : BaseQuickAdapter<ArticleBean, BaseViewHolder>(R.layout.item_article, data) {

    override fun onItemViewHolderCreated(viewHolder: BaseViewHolder, viewType: Int) {
        DataBindingUtil.bind<ViewDataBinding>(viewHolder.itemView)
    }

    override fun convert(holder: BaseViewHolder, item: ArticleBean) {
        val binding = holder.getBinding<ItemArticleBinding>()
        if (binding != null) {
            binding.articleBean = item
        }
    }
}